entrypoint.sh

```diff
@@ -27,6 +27,18 @@ elif [ "$MODE" == "worker" ] ; then
     /opt/spark/sbin/start-worker.sh --cores "$CORES" --memory "$MEM" "$SPARK_MASTER_URL"
 elif [ "$MODE" == "bash" ] ; then
     bash
+elif [ "$MODE" == "notebook" ] ; then
+    echo "starting jupyter notebook"
+    WORKSPACE_DIR="/cdm_shared_workspace"
+    mkdir -p "$WORKSPACE_DIR" || { echo "Error: Failed to create workspace directory"; exit 1; }
+    cd "$WORKSPACE_DIR" || { echo "Error: Failed to navigate to workspace directory"; exit 1; }
+    jupyter lab --ip=0.0.0.0 \
+                --port=$PORT \
+                --no-browser \
+                --allow-root \
+                --notebook-dir="$WORKSPACE_DIR" \
+                --ServerApp.token='' \
+                --ServerApp.password=''
```

Comment:
How does the notebook / pyspark know where the local spark installation lives?

The driver will have to be configured to advertise the host, port, and bindManager port, e.g.:

```
root@f058c872158d:/opt/spark# bin/spark-submit --master spark://10.58.1.104:7077 --conf spark.driver.bindAddress=0.0.0.0 --conf spark.driver.host=$SPARK_DRIVER_HOST --conf spark.driver.port=$SPARK_DRIVER_PORT --conf spark.blockManager.port=$SPARK_BLOCKMANAGER_PORT examples/src/main/python/pi.py 10 2>/dev/null
Pi is roughly 3.144080
```
The ports will need to be accessible to the workers

Reply:
In the command above the spark master url should be an env var as well

Reply:
added `SPARK_MASTER_URL` under environment. I was manually configuring the spark master url before.